app/src/main/java/com/greenfox/kalendaryo/SelectCalendarActivity.java

```diff
@@ -87,23 +99,47 @@ public void getCalendarList() {
         ArrayList<String> accounts = kalPref.getAccounts();
 
         for (int i = 0; i < accounts.size(); i++) {
-            GoogleAuth googleAuth = kalPref.getAuth(accounts.get(i));
-
-            String accessToken = googleAuth.getAccessToken();
-            String authorization = "Bearer " + accessToken;
-
-            googleApi.getCalendarList(authorization).enqueue(new Callback<GoogleCalendarsResponse>() {
-                @Override
-                public void onResponse(Call<GoogleCalendarsResponse> call, Response<GoogleCalendarsResponse> response) {
+            String account = accounts.get(i);
+            requestCalendars(account, FIRST_ATTEMPT);
+        }
+    }
+    public void requestCalendars (String account, Integer attempt) {
+        GoogleAuth googleAuth = kalPref.getAuth(account);
+        String authorization = "Bearer " + googleAuth.getAccessToken();
+        googleApi.getCalendarList(authorization).enqueue(new Callback<GoogleCalendarsResponse>() {
+            @Override
+            public void onResponse(Call<GoogleCalendarsResponse> call, Response<GoogleCalendarsResponse> response) {
+                if (response.errorBody() == null) {
                     adapter.addGoogleCalendars(response.body().getItems());
                     googleCalendars.addAll(response.body().getItems());
+                } else if (attempt != FINAL_ATTEMPT){
+                    requestAccessTokenRefresh(googleAuth, kalPref.clientToken());
+                    requestCalendars(account, FINAL_ATTEMPT);
```

Comment:
should be `attempt++; requestCalendars(account, attempt);`, so if we change the FINAL_ATTEMPT to be greater than 2 it will still work as expected